src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -147,12 +156,22 @@ protected override void ApplyResponseChallenge()
                 properties = new AuthenticationProperties(ChallengeContext.Properties);
             }
 
-            if (string.IsNullOrEmpty(properties.RedirectUri))
+            if (!string.IsNullOrWhiteSpace(properties.RedirectUri))
             {
+                _logger.LogDebug(format: Resources.OIDCH_0030_Using_Properties_RedirectUri, args: properties.RedirectUri);
+            }
+            else if (Options.DefaultToCurrentUriOnRedirect)
```

Comment:
Is `DefaultToCurrentUriOnRedirect` still useful? I'd personally remove it and bring back the old behavior (i.e always use `properties.RedirectUri` when present).